### PR TITLE
[UT] add testcases for url_extract_host function

### LINKED ISSUE #31969: Need support url_extract_host function in builtfunction
## Feature request

**Is your feature request related to a problem? Please describe.**
starrrocks need url_extract_host function in built function 

**Describe the solution you'd like**
pr as https://github.com/StarRocks/starrocks/pull/31968

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by xuzifu666**
Fixes [#31969](https://github.com/StarRocks/starrocks/issues/31969)

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [x] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 19 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exprs/string_functions.cpp|0|19|00.00%|[2518, 2519, 2520, 2521, 2522, 2523, 2524, 2527, 2528, 2530, 2531, 2532, 2533, 2535, 2537, 2540, 2541, 2544, 2545]|

File: test/sql/test_string_functions/R/test_string_functions [hunk 1]
```diff
@@ -236,6 +236,24 @@ select split_part(c1, "##", -3) from t1;
...
+-- !result
+select url_extract_host('https://starrocks.快速.com/test/api/v1');
+-- result:
+starrocks.快速.com
+-- !result
+select url_extract_host(NULL);
+-- result:
+None
+-- !result
+select url_extract_host('https://starrocks.com/doc?k1=10&k2=3&k1=100');
```
Comment (kangkaisen) [inline test/sql/test_string_functions/R/test_string_functions:252]: What if the argument of url_extract_host is string column, not the const string?

Comment (xuzifu666) [reply test/sql/test_string_functions/R/test_string_functions:252]: > What if the argument of url_extract_host is string column, not the const string?

make sense，i would add some testcase for it @kangkaisen

**Review by kangkaisen — DISMISSED** [review]

**Comment by superdiaodiao** [comment]
Why did you use my pr code into your pr without my permission? https://github.com/StarRocks/starrocks/pull/31990

**Comment by xuzifu666** [comment]
> Why did you use my pr code into your pr without my permission? #31990
issue and pr had assgine to me firstly，some good idea you can suggest in the pre pr；you write the same the testcase、doc without comunication with me too @superdiaodiao

**Comment by xuzifu666** [comment]
> > Why did you use my pr code into your pr without my permission? #31990
> 
> issue and pr had assgine to me firstly，some good idea you can suggest in the pre pr；you write the same the testcase、doc in this pr without comunication with me is not necessary@superdiaodiao

**Comment by superdiaodiao** [comment]
> > Why did you use my pr code into your pr without my permission? #31990
> 
> issue and pr had assgine to me firstly，some good idea you can suggest in the pre pr；you write the same the testcase、doc without comunication with me@superdiaodiao 

I don't think the pr only belongs to you, but the way you copy my develop code is nonsence. By the way, the testcase is not the same.

**Comment by xuzifu666** [comment]
> > > Why did you use my pr code into your pr without my permission? #31990
> > 
> > 
> > issue and pr had assgine to me firstly，some good idea you can suggest in the pre pr；you write the same the testcase、doc without comunication with me@superdiaodiao
> 
> I don't think the pr only belongs to you, but the way you copy my develop code is nonsence. By the way, the testcase is not the same.

firstly，this is not only you can use the method to resolve the problem，method is same mean copy？

**Comment by superdiaodiao** [comment]
> > > > Why did you use my pr code into your pr without my permission? #31990
> > > 
> > > 
> > > issue and pr had assgine to me firstly，some good idea you can suggest in the pre pr；you write the same the testcase、doc without comunication with me@superdiaodiao
> > 
> > I don't think the pr only belongs to you, but the way you copy my develop code is nonsence. By the way, the testcase is not the same.
> 
> firstly，this is not only you can use the method to resolve the problem，method is same mean copy？

Why did you use the same develop code after me? I think this behavior is inproper.

**Comment by xuzifu666** [comment]
> I


testcase and doc only just  same with the pr nearly after I submit issue and pr，it also means copy without my permission？this behavor is proper？

**Comment by xuzifu666** [comment]
> > > > > Why did you use my pr code into your pr without my permission? #31990
> > > > 
> > > > 
> > > > issue and pr had assgine to me firstly，some good idea you can suggest in the pre pr；you write the same the testcase、doc without comunication with me@superdiaodiao
> > > 
> > > 
> > > I don't think the pr only belongs to you, but the way you copy my develop code is nonsence. By the way, the testcase is not the same.
> > 
> > 
> > firstly，this is not only you can use the method to resolve the problem，method is same mean copy？
> 
> Why did you use the same develop code after me? I think this behavior is inproper.



testcase and doc only just  same with the pr nearly after I submit issue and pr，it also means copy without my permission？this behavor is proper？

**Comment by superdiaodiao** [comment]
> > > > > > Why did you use my pr code into your pr without my permission? #31990
> > > > > 
> > > > > 
> > > > > issue and pr had assgine to me firstly，some good idea you can suggest in the pre pr；you write the same the testcase、doc without comunication with me@superdiaodiao
> > > > 
> > > > 
> > > > I don't think the pr only belongs to you, but the way you copy my develop code is nonsence. By the way, the testcase is not the same.
> > > 
> > > 
> > > firstly，this is not only you can use the method to resolve the problem，method is same mean copy？
> > 
> > Why did you use the same develop code after me? I think this behavior is inproper.
> 
> 
> 
> testcase and doc only just  same with the pr nearly after I submit issue and pr，it also means copy without my permission？this behavor is proper？

Again, testcase and doc is not the same, ok? The develop code is the matter.

If so, why don't you copy others' develop code into your pr just like you did the works?

**Comment by xuzifu666** [comment]
> > > > > > > Why did you use my pr code into your pr without my permission? #31990
> > > > > > 
> > > > > > 
> > > > > > issue and pr had assgine to me firstly，some good idea you can suggest in the pre pr；you write the same the testcase、doc without comunication with me@superdiaodiao
> > > > > 
> > > > > 
> > > > > I don't think the pr only belongs to you, but the way you copy my develop code is nonsence. By the way, the testcase is not the same.
> > > > 
> > > > 
> > > > firstly，this is not only you can use the method to resolve the problem，method is same mean copy？
> > > 
> > > 
> > > Why did you use the same develop code after me? I think this behavior is inproper.
> > 
> > 
> > testcase and doc only just  same with the pr nearly after I submit issue and pr，it also means copy without my permission？this behavor is proper？
> 
> Again, testcase and doc is not the same, ok? The develop code is the matter.
> 
> If so, why don't you copy others' develop code into your pr just like you did the works?

Everything you said is right，best wish to you

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exprs/string_functions.cpp|2|6|33.33%|[2520, 2521, 2522, 2523]|

**Comment by kateshaowanjou** [comment]
@superdiaodiao I think there might be some misunderstanding in this. Can you give me your contact information so we can chat?

**Comment by superdiaodiao** [comment]
> @superdiaodiao I think there might be some misunderstanding in this. Can you give me your contact information so we can chat?

Thanks for asking, I have contact you via Slack.

By the way, a few days ago, @alberttwong knew my contribution which means an earlier code version.

**Comment by xuzifu666** [comment]
close the pr and would submit a testcase pr for it

**Review by kangkaisen — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exprs/string_functions.cpp|2|3|66.67%|[3969]|

**Pull request merged**